pycbc/detector.py

```diff
@@ -444,3 +445,138 @@ def time_delay_from_earth_center(self, right_ascension, declination, t_gps):
             np.array([np.float32(earth.x), np.float32(earth.y),
                       np.float32(earth.z)]), right_ascension,
             declination, t_gps)
+
+    def interpolate_hij(self, hp, hc):
+        """Returns the values of hp, hc after
+            interpoalting it to the desired time
+        Parameters
+        ----------
+        ref_time : numpy.ScalarType
+            GPS time
+        hp, hc : pycbc.types.TimeSeries
+            Plus and cross polarizations in order
+        time : numpy.ndarray (3,1)
+            GPS time
+        Returns
+        -------
+        numpy.ndarray shape (3,1)
+            The values of hij at the required time
+        """
+        fhp = interpolate.interpolate.interp1d(hp.sample_times, hp)
+        fhc = interpolate.interpolate.interp1d(hc.sample_times, hc)
+        xnew = np.linspace(hp.sample_times[0], hp.sample_times[-1], len(hp)*100)
+        hp, hc = fhp(xnew), fhc(xnew)
+        fhp = TimeSeries(hp, delta_t=1/len(hp))
+        fhc = TimeSeries(hc, delta_t=1/len(hc))
+        return fhp, fhc, xnew
+
+    def eval_hij(self, hp, hc, k, ref_time, t):
+        """Returns the values of hij
+        Parameters
+        ----------
+        hp, hc : pycbc.types.TimeSeries
+            Plus and cross polarizations in order
+        k : numpy.ndarray shape(3,1)
+            Sky-loaction in Cartesian system
+        ref_time : numpy.ScalarType
+            GPS time
+        time : numpy.ScalarType
+            GPS time
+        Returns
+        -------
+        numpy.ndarray shape (3,1)
+            The values of hij at the required time
+        """
+        k = k/(k[0]**2 + k[1]**2 + k[2]**2)**.5
```

Comment:
It seem you bust convert back directly to sky angles, so why not have those be the input arguments directly?